include/zephyr/senss/senss.h

```diff
@@ -0,0 +1,522 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef SENSS_H_
+#define SENSS_H_
+
+/**
+ * @defgroup senss Sensor Subsystem
+ * @defgroup senss_sensor_types  Sensor Types
+ * @ingroup senss
+ * @defgroup senss_datatypes Data Types
+ * @ingroup senss
+ * @defgroup senss_sensor Sensor API
+ * @ingroup senss
+ * @defgroup senss_sensor_callbacks Sensor Callbacks
+ * @ingroup senss_sensor
+ * @defgroup senss_physical Physical Sensor API
+ * @ingroup senss_sensor
+ * @defgroup senss_virtual Virtual Sensor API
+ * @ingroup senss_sensor
+ */
+
+#include <senss/senss_datatypes.h>
+#include <senss/senss_sensor_types.h>
+#include <senss/senss_sensor.h>
+
+/**
+ * @brief Sensor Subsystem API
+ * @addtogroup senss_api
+ * @{
+ */
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Initialize sensor subsystem core.
+ *
+ * This will allocate internal resources, such as create internal schedule
+ * task, create internal management data structures.
+ *
+ * It will also enumerate all sensor instances which registered by
+ * \ref SENSS_REGISTER_SENSOR, build sensors report relational data
+ * model, if a virtual sensor's required dependency sensor type not found, will
+ * return error.
+ * Then call each sensor instances' init callback API.
+ *
+ * @return 0 on success or negative error value on failure.
+ *
+ */
+int senss_init(void);
+
+ /**
+  * @brief De-initialize sensor subsystem core.
+  *
+  * This will release all dynamically allocated internal resources, and call
+  * all sensor instances' deinit callback API.
+  *
+  * @return 0 on success or negative error value on failure.
+  */
+int senss_deinit(void);
+
+ /**
+  * @brief Find sensor instances by sensor type and instance ID.
+  *
+  * Application can use these API to find sensor instances information by
+  * sensor types and instance id, or enumerate all sensor instances'
+  * information at one call.
+  *
+  * This API just returned read only information of sensor instances,
+  * no side effect to sensor instances.
+  *
+  * @param type The sensor type which need to find,
+  * \ref SENSS_SENSOR_TYPE_ALL for all types.
+  *
+  * @param instance The instance ID which need to find, -1 for find all
+  * instances with type \p type
+  *
+  * @param max The max count of the \p infos array input. Can get real count
+  * number via \ref senss_get_sensor_count.
+  *
+  * @param *info Input info array for receiving found sensor instances
+  *
+  * @return Number of sensor instances found, 0 returned if no found
+  */
+int senss_find_sensor(int type, int instance,
+		 int max, struct senss_sensor_info *info);
```

Comment:
Yes,  this function is useful, and can easy to support `CHRE PAL`,  such as `getSensors`,  `chreSensorFind`. 

Its also useful for no `CHRE` configured options, for example,  Application can used it easily enumerate sensor instance by type and instance,  all enumerate all. 

The reason of why not stick to something more closer to the `PAL`,  because the sensor types definitions and sensor information structured definitions,  since `sensor subsystem`  targets to support cross OSes (with extra transport layers),  except `CHRE` and `Chrome` support,  also need consider how to easy support `HID` and Windows.  The problem is `CHRE` and `HID` has different sensor types and information definitions,  stick to one closer,  will harder to support other.  

`sensor subsystem` will try to find a superset definitions to easy support both `CHRE` and `HID`.